### ISSUE TITLE
Additivity check failed in GPUTreeExplainer but works fine in TreeExplainer with Light GBM

### ISSUE THREAD

**Issue by normanwang92** [shap/shap#1852]
shap version 0.38.1
both training set and validation set are float64 dataframes
![image](https://user-images.githubusercontent.com/11740145/109897905-608ea380-7cce-11eb-8abf-c4aa03251eeb.png)
